### PR TITLE
ROX-22101: use gogo TimestampFromProto through abstraction layer

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

The goal is to remove direct calls to the `gogo` protobuf layer and have all of them happen through an abstraction layer in order to ease the migration to a new protobuf library.

The change is the application through the refactoring tool `eg` of the following template:
```
package refactor

import (
	"time"

	"github.com/gogo/protobuf/types"
	"github.com/stackrox/rox/pkg/protocompat"
)

func before(ts *types.Timestamp) (time.Time, error) {
	return types.TimestampFromProto(ts)
}

func after(ts *types.Timestamp) (time.Time, error) {
	return protocompat.ConvertTimestampToTimeOrError(ts)
}
```

PR stack:
- #9148
  - #9156 
  - #9145 
  - #9196 
  - #9198
  - #9201

## Checklist
- [ ] Investigated and inspected CI test results
~~- [ ] Unit test and regression tests added~~
~~- [ ] Evaluated and added CHANGELOG entry if required~~
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by vikin91 — COMMENTED** [review]
Great idea - thank you for starting working on this!
I think CI is unhappy with the changes - could you give this a look?

**Review by janisz — APPROVED** [review]
Nice first step. There are some helper functions that still rely on gogo types but we can handle them in next PR

File: pkg/protoconv/time.go [hunk 1]
```diff
@@ -23,6 +23,11 @@ func ConvertGoGoProtoTimeToGolangProtoTime(gogo *gogoTimestamp.Timestamp) *golan
 	}
 }
 
+// ConvertTimestampToTimeOrError converts a proto timestamp to a golang Time, or returns an error if there is one.
+func ConvertTimestampToTimeOrError(gogo *gogoTimestamp.Timestamp) (time.Time, error) {
```
Comment (RTann) [inline pkg/protoconv/time.go:27]: gotta say I'm not a huge fan of these Java-like names

File: pkg/protoconv/time.go
Comment (RTann) [inline pkg/protoconv/time.go:1]: I'm hoping you can help me understand this more. I like the idea of abstracting out the specific proto type, but it seems like this abstraction layer still depends a lot on gogoproto for inputs and outputs. I feel like we'd need to first stop depending on the gogoproto in favor of some time interface that we use instead and then use gogoproto under the hood in this abstraction layer. That should limit the gogoproto usage to just here and then make it easy to switch proto types whenever we want. That may be a lot easier said than done, though

**Review by md2119 — APPROVED** [review]

File: pkg/protoconv/time.go
Comment (janisz) [reply pkg/protoconv/time.go:1]: I think the goal of protocompat layer is to have a single place where we need to do a change if we migrate out of gogoproto.

File: central/pruning/pruning.go
Comment (mtodor) [inline central/pruning/pruning.go:1]:
```
"github.com/gogo/protobuf/types"
```
is not used anymore. We should remove it.

**Review by mtodor — APPROVED** [review]
Looks good. There is a small leftover.

File: pkg/protoconv/time.go (hunk 1 above)
Comment (mtodor) [reply pkg/protoconv/time.go]: When we finish with proto migration to V2, we will have a cleanup phase to remove all introduced conversion functions. That will be a chance to rename (or remove) such functions.

**Review by mtodor — APPROVED** [review]

**Pull request merged**